controllers/gateway/api_controller_integration_test.go

```diff
@@ -1948,6 +1951,295 @@ var _ = Describe("APIRule Controller", Serial, func() {
+				g.Expect(expectedApiRule.Status.APIRuleStatus.Code).To(Equal(gatewayv1beta1.StatusOK))
+				g.Expect(expectedApiRule.ObjectMeta.Annotations).To(HaveKeyWithValue("gateway.kyma-project.io/original-version", "v1beta1"))
+				g.Expect(expectedApiRule.ObjectMeta.Annotations).To(HaveKey("gateway.kyma-project.io/v1beta1-spec"))
+
+				ruleV1beta1 := gatewayv1beta1.APIRule{}
+				err := json.Unmarshal([]byte(expectedApiRule.Annotations["gateway.kyma-project.io/v1beta1-spec"]), &ruleV1beta1.Spec)
+				g.Expect(err).To(BeNil())
+				g.Expect(expectedApiRule.Spec).To(Equal(ruleV1beta1.Spec))
+			}, eventuallyTimeout).Should(Succeed())
+
+			By("Verifying APIRule fetched in version v2alpha1")
+			Eventually(func(g Gomega) {
+				expectedApiRule := gatewayv2alpha1.APIRule{}
+				g.Expect(c.Get(context.Background(), client.ObjectKey{Name: apiRuleName, Namespace: testNamespace}, &expectedApiRule)).Should(Succeed())
+				g.Expect(expectedApiRule.Status.State).To(Equal(gatewayv2alpha1.Ready))
+				g.Expect(expectedApiRule.ObjectMeta.Annotations).To(HaveKeyWithValue("gateway.kyma-project.io/original-version", "v1beta1"))
+				g.Expect(expectedApiRule.ObjectMeta.Annotations).To(HaveKey("gateway.kyma-project.io/v1beta1-spec"))
+
+				ruleV1beta1 := gatewayv1beta1.APIRule{}
+				err := json.Unmarshal([]byte(expectedApiRule.Annotations["gateway.kyma-project.io/v1beta1-spec"]), &ruleV1beta1.Spec)
+				g.Expect(err).To(BeNil())
+				g.Expect(string(*expectedApiRule.Spec.Hosts[0])).To(Equal(*ruleV1beta1.Spec.Host))
+				g.Expect(expectedApiRule.Spec.Gateway).To(Equal(ruleV1beta1.Spec.Gateway))
+				g.Expect(expectedApiRule.Spec.Rules).To(HaveLen(len(ruleV1beta1.Spec.Rules)))
+			}, eventuallyTimeout).Should(Succeed())
+
+			By("Verifying APIRule fetched in version v2")
+			Eventually(func(g Gomega) {
+				expectedApiRule := gatewayv2.APIRule{}
+				g.Expect(c.Get(context.Background(), client.ObjectKey{Name: apiRuleName, Namespace: testNamespace}, &expectedApiRule)).Should(Succeed())
+				g.Expect(expectedApiRule.Status.State).To(Equal(gatewayv2.Ready))
+				g.Expect(expectedApiRule.ObjectMeta.Annotations).To(HaveKeyWithValue("gateway.kyma-project.io/original-version", "v1beta1"))
+				g.Expect(expectedApiRule.ObjectMeta.Annotations).To(HaveKey("gateway.kyma-project.io/v1beta1-spec"))
+
+				ruleV1beta1 := gatewayv1beta1.APIRule{}
+				err := json.Unmarshal([]byte(expectedApiRule.Annotations["gateway.kyma-project.io/v1beta1-spec"]), &ruleV1beta1.Spec)
+				g.Expect(err).To(BeNil())
+				g.Expect(string(*expectedApiRule.Spec.Hosts[0])).To(Equal(*ruleV1beta1.Spec.Host))
+				g.Expect(expectedApiRule.Spec.Gateway).To(Equal(ruleV1beta1.Spec.Gateway))
+				g.Expect(expectedApiRule.Spec.Rules).To(HaveLen(len(ruleV1beta1.Spec.Rules)))
+			}, eventuallyTimeout).Should(Succeed())
+		})
+
+		It("should not fetch the APIRule when v1beta1 is the original-version and the spec is not convertible", func() {
```

Comment:
Naming is not accurate in that case
```suggestion
		It("should fetch the empty spec for APIRule v2alpha1 and v2 when v1beta1 is the original-version and spec is not convertible", func() {
```